### DISCUSSION TITLE
How Zustand works under the hood ?

### DISCUSSION THREAD

**Discussion by AhmadSaman** [pmndrs/zustand#1832] [Q&A] [answered]
Hi there, lately I have been enjoying Zustand, but every time I was using it in a project I always asked myself **How this works without Providers ?** after searching didn't get any clear answer, anyone out there can give me an answer

How do Zustand stores states without wrapping my react project with a Provider? what is the approach?

**Comment by dai-shi** [comment, marked as answer]
I created a course with that perspective: https://egghead.io/courses/create-a-zustand-like-module-state-library-bf55241e
And also this: https://daishi.gumroad.com/l/uaxms

**Reply by mr-rime** [reply]
Thank you that's really cool 👌

**Comment by yshterev** [comment]
@dai-shi Thank you! Awesome tutorial 🚀
